### PR TITLE
feat: add settings functionality

### PR DESCRIPTION
**Pull request by Shurtu-gal**
### Description
- Have added custom settings for prettier.
- Integrated eslint with prettier
- Added tests for prettier

### Fixes
closes amplication/amplication#6606
closes amplication/amplication#6709

### CODE REVIEW

**Comment by overbit** [comment]
@Shurtu-gal can you please add some unit test to cover the expected results?

**Comment by Shurtu-gal** [comment]
Sure I would do so..

**Comment by Shurtu-gal** [comment]
@overbit just had a tiny doubt regarding how to mock importStaticModules in the tests?

**Comment by overbit** [comment]
@Shurtu-gal you could avoid mocking it and use the real method for it. Any thoughts on this option?

**Comment by Shurtu-gal** [comment]
> @Shurtu-gal you could avoid mocking it and use the real method for it. Any thoughts on this option?

Can you tell me where to import it from?

**Comment by overbit** [comment]
> > @Shurtu-gal you could avoid mocking it and use the real method for it. Any thoughts on this option?
> 
> Can you tell me where to import it from?

Answered offline.

**Comment by overbit** [comment]
@Shurtu-gal, once this is completed, would you be able to open a new PR for tests in the eslint too?

File: plugins/formatter-prettier/.amplicationrc.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "settings": {
+    "rules": {
+      "singleQuote": true,
+      "arrowParens": "avoid",
+      "trailingComma": "all"
```
Comment (overbit) [inline plugins/formatter-prettier/.amplicationrc.json:6]: Let's use the default rules `{}`

**Review by overbit — CHANGES_REQUESTED** [review]

File: plugins/formatter-prettier/.amplicationrc.json (hunk 1 above)
Comment (Shurtu-gal) [reply plugins/formatter-prettier/.amplicationrc.json:6]: Okay

Comment (Shurtu-gal) [reply plugins/formatter-prettier/.amplicationrc.json]: @overbit Done

**Review by overbit — APPROVED** [review]
Nice work @Shurtu-gal!

**Pull request merged**